packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -103,17 +120,41 @@ function DocSidebarItemCategory({
         'menu__list-item--collapsed': collapsed,
       })}
       key={label}>
-      <a
-        className={clsx('menu__link', {
-          'menu__link--sublist': collapsible,
-          'menu__link--active': collapsible && isActive,
-          [styles.menuLinkText]: !collapsible,
-        })}
-        onClick={collapsible ? handleItemClick : undefined}
-        href={collapsible ? '#!' : undefined}
-        {...props}>
-        {label}
-      </a>
+      {goToLink && collapsed ? (
+        <Link
+          to={intitalHref}
+          className={clsx('menu__link', {
+            'menu__link--sublist': collapsible,
+            'menu__link--active': collapsible && isActive,
+            [styles.menuLinkText]: !collapsible,
+          })}
+          {...(isInternalUrl(intitalHref)
+            ? {
+                isNavLink: true,
+                exact: true,
+                onClick: onItemClick,
+              }
+            : {
+                target: '_blank',
+                rel: 'noreferrer noopener',
+              })}
+          onClick={collapsible ? handleItemClick : undefined}
+          {...props}>
+          {label}
+        </Link>
+      ) : (
+        <a
+          className={clsx('menu__link', {
+            'menu__link--sublist': collapsible,
+            'menu__link--active': collapsible && isActive,
+            [styles.menuLinkText]: !collapsible,
+          })}
+          onClick={collapsible ? handleItemClick : undefined}
+          href={collapsible ? '#!' : undefined}
+          {...props}>
+          {label}
+        </a>
+      )}
```

Comment:
That's looks like quite a lot of code duplication.
Can't we just `<Link>` in both cases (category is a link, or is not a link)?
You can probably call `e.preventDefault()` when the category does not  link to anywhere and  acts like a button (like it used to be)

Reply:
I agree, I made this change.